src/components/Carousel.js

```diff
@@ -0,0 +1,250 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {omit} from 'ramda';
+
+import {
+  Carousel as RSCarousel,
+  CarouselItem,
+  CarouselControl,
+  CarouselIndicators,
+  CarouselCaption
+} from 'reactstrap';
+
+/**
+ * Component for creating Bootstrap carousel.  This component is a slideshow
+ * for cycling through a series of content.
+ */
+const Carousel = props => {
+  const [animating, setAnimating] = useState(false);
+  const {
+    items,
+    active_index,
+    style,
+    className,
+    loading_state,
+    controls,
+    indicators,
+    setProps,
+    ...otherProps
+  } = props;
+
+  const next = () => {
+    if (animating) return;
+    const nextIndex = active_index === items.length - 1 ? 0 : active_index + 1;
+    setProps({active_index: nextIndex});
+  };
+
+  const previous = () => {
+    if (animating) return;
+    const nextIndex = active_index === 0 ? items.length - 1 : active_index - 1;
+    setProps({active_index: nextIndex});
+  };
+
+  const goToIndex = newIndex => {
+    if (animating) return;
+    setProps({active_index: newIndex});
+  };
+
+  const slides = items.map(item => {
+    // note - the default 'd-block w-100' is from the examples in the Bootstrap docs.
+    item.imgClassName =
+      typeof item.imgClassName !== 'undefined'
+        ? item.imgClassName
+        : 'd-block w-100';
+
+    return (
+      <CarouselItem
+        onExiting={() => setAnimating(true)}
+        onExited={() => setAnimating(false)}
+        key={item.id}
```

Comment:
What do you think about renaming the `item` `id` prop to `key`?   I named it `id` because that's more commonly used in Dash, but people might think you can target a slide number in a callback using this `id`. (You can't)

Reply:
Yeah, if it's only being used as a `key` then I think renaming makes sense. Perhaps you can make it optional and set a default?

Something like (untested)

```js
const slides = items.map((item, idx) => {
    const key = item.key || `_dbc-carouselitem-${idx}`;
    // ...
});
```